### PR TITLE
feat(homepage): Add recordings table to homepage

### PR DESCRIPTION
**Pull request by rcmarron**
## Changes

This PR mainly preps the recordings table component to be added to the home page.

The big changes are:
* Adds a `onlyShowList` mode to the recordings table that:
    * Removes the filters from above the recordings list
    * Stops the recordings list from adding filters to the page's URL
* A bit of clean up to move the recording filters to the hashParams from query params
* Added the recording table to the homepage. Note: still more work to do here, but this page is behind a FF `project-homepage`.

*Stay up-to-date with our [coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*
Screen Shot 2022-03-24 at 11 25 12 AM

## How did you test this code?

Added unit tests for the homepage case

### CODE REVIEW

File: frontend/src/scenes/project-homepage/ProjectHomepage.tsx [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import React from 'react'
 import './ProjectHomepage.scss'
 import { PageHeader } from 'lib/components/PageHeader'
+import { SessionRecordingsTable } from 'scenes/session-recordings/SessionRecordingsTable'
+import { RecordingTableLocation } from 'scenes/session-recordings/sessionRecordingsTableLogic'
```
Comment (alexkim205) [inline frontend/src/scenes/project-homepage/ProjectHomepage.tsx:6]: I think we need to export a scene object here

File: frontend/src/scenes/session-recordings/SessionPlayerDrawer.tsx [hunk 2]
```diff
@@ -29,7 +28,7 @@ export function SessionPlayerDrawer({ isPersonPage = false, onClose }: SessionPl
                     justify="space-between"
                 >
                     <Button type="link" onClick={onClose}>
-                        <ArrowLeftOutlined /> Back to {isPersonPage ? 'persons' : 'recordings'}
+                        <ArrowLeftOutlined /> Back
```
Comment (alexkim205) [inline frontend/src/scenes/session-recordings/SessionPlayerDrawer.tsx:31]: Nice I like the generalization here. Thought this was too specific in the first place.

File: frontend/src/scenes/session-recordings/sessionRecordingsTableLogic.ts [hunk 3]
```diff
@@ -55,11 +58,15 @@ export const DEFAULT_ENTITY_FILTERS = {
     ],
 }
 
-export const sessionRecordingsTableLogic = kea<sessionRecordingsTableLogicType<PersonUUID>>({
+export const DEFAULT_FROM_DATE = '-21d'
+
+export const sessionRecordingsTableLogic = kea<sessionRecordingsTableLogicType<PersonUUID, RecordingTableLocation>>({
     path: (key) => ['scenes', 'session-recordings', 'sessionRecordingsTableLogic', key],
-    key: (props) => props.personUUID || 'global',
+    key: (props) => props.personUUID || props.tableLocation || 'global',
```
Comment (alexkim205) [inline frontend/src/scenes/session-recordings/sessionRecordingsTableLogic.ts:65]: Can this align with the pageKey defined in the component?

File: frontend/src/scenes/session-recordings/sessionRecordingsTableLogic.ts [hunk 4]
```diff
@@ -246,57 +253,65 @@ export const sessionRecordingsTableLogic = kea<sessionRecordingsTableLogicType<P
...
             }
         ] => {
-            const params: Params = {
-                filters: values.filterQueryParams,
-                ...overrides,
-            }
             const hashParams: HashParams = {
                 ...router.values.hashParams,
             }
+            console.log(props, Object.keys(values.filterQueryParams).length === 0)
```
Comment (alexkim205) [inline frontend/src/scenes/session-recordings/sessionRecordingsTableLogic.ts:271]: leftovers

**Review by alexkim205 — CHANGES_REQUESTED** [review]

**Comment by EDsCODE** [comment]
Follow up:

- Make the empty table more encouraging to users who might not have recordings enabled. 
- Allow people to remove that module if it's useless to them

**Pull request closed without merging**